src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityFissionReactor.java

```diff
@@ -0,0 +1,1104 @@
+        this.fuelDepletionPercent = Math.max(0, this.fissionReactor.fuelDepletion) /
+                this.fissionReactor.maxFuelDepletion;
+        writeCustomData(GregtechDataCodes.SYNC_REACTOR_STATS, (packetBuffer -> {
+            packetBuffer.writeDouble(this.fissionReactor.temperature);
+            packetBuffer.writeDouble(this.fissionReactor.maxTemperature);
+            packetBuffer.writeDouble(this.fissionReactor.pressure);
+            packetBuffer.writeDouble(this.fissionReactor.maxPressure);
+            packetBuffer.writeDouble(this.fissionReactor.power);
+            packetBuffer.writeDouble(this.fissionReactor.maxPower);
+            packetBuffer.writeDouble(this.fissionReactor.kEff);
+            packetBuffer.writeDouble(this.fissionReactor.controlRodInsertion);
+            packetBuffer.writeDouble(this.fuelDepletionPercent);
+        }));
+        this.markDirty();
+    }
+
+    @Override
+    public void receiveCustomData(int dataId, PacketBuffer buf) {
+        super.receiveCustomData(dataId, buf);
+
+        if (dataId == GregtechDataCodes.SYNC_REACTOR_STATS) {
+            this.temperature = buf.readDouble();
+            this.maxTemperature = buf.readDouble();
+            this.pressure = buf.readDouble();
+            this.maxPressure = buf.readDouble();
+            this.power = buf.readDouble();
+            this.maxPower = buf.readDouble();
+            this.kEff = buf.readDouble();
+            this.controlRodInsertionValue = buf.readDouble();
+            this.fuelDepletionPercent = buf.readDouble();
+        } else if (dataId == GregtechDataCodes.SYNC_LOCKING_STATE) {
+            this.lockingState = buf.readEnumValue(LockingState.class);
+            this.scheduleRenderUpdate();
+        }
+    }
+
+    protected void lockAll() {
+        for (ICoolantHandler handler : this.getAbilities(MultiblockAbility.IMPORT_COOLANT)) {
+            handler.setLock(true);
+        }
+        for (IFuelRodHandler handler : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+            handler.setLock(true);
+        }
+    }
+
+    protected void unlockAll() {
+        for (ICoolantHandler handler : this.getAbilities(MultiblockAbility.IMPORT_COOLANT)) {
+            handler.setLock(false);
+        }
+        for (IFuelRodHandler handler : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+            handler.setLock(false);
+        }
+        if (this.fissionReactor != null) {
+            this.fissionReactor.turnOff();
+        }
+        if (this.lockingState == LockingState.LOCKED) { // Don't remove warnings
+            this.setLockingState(LockingState.UNLOCKED);
+        }
+    }
+
+    private void lockAndPrepareReactor() {
+        this.lockAll();
+        int radius = this.diameter / 2;     // This is the floor of the radius, the actual radius is 0.5 blocks
+        // larger
+        BlockPos reactorOrigin = this.getPos().offset(this.frontFacing.getOpposite(), radius);
```

Comment:
Use a MutableBlockPos instead for this method. Offset creates a new blockpos every time.